### ISSUE TITLE
Possibly add concurrent test running

### ISSUE THREAD

**Issue by BenLambertNcl** [symphonyoss/ContainerJS#164]
It would be good if we could get the test suites to run concurrently. Currently it takes >15 minutes to run the tests on each PR and merge. This could be almost halved if we could get the tests to run concurrently.

**Comment by jleft** [comment]
Definitely would be good to speed up the tests 👍 

If I understand correctly, AppVeyor supports [parallel testing](https://www.appveyor.com/docs/parallel-testing/), which distributes test across multiple VMs, but we're limited to only *1* concurrent build, unless we upgrade to a paid account (which is a shame; Travis supports *200!*). [Rolling Builds](https://www.appveyor.com/docs/build-configuration/#rolling-builds) may help if we have a long build queue.

We could setup something to run the tests concurrently on a single VM (I think the build servers will give us access to 2 cores, so there's scope to parallelise), but I'm not sure if Spectron will work in parallel - if it does, then that would be great!

Even if we do parallelise the tests, I think at best on AppVeyor we'll half the time - and the tests will take even longer as we add more. Could be worth considering running the full test suite on `master` and only running a limited set of tests on pull requests.

**Comment by BenLambertNcl** [comment]
I haven't looked too much into AppVeyor so thanks for that :). I have been trying to get spectron to run locally in parallel with little success, although it does seem like webdriverio/chrome driver both support parallel testing, so it seems like it should work. 

I was also looking at changing the tests so we don't restart the electron/openfin instance after each test. This seems to improve performance, but theres a possibility that some tests can affect other tests (although this can me mitigated by reloading the page after each test, not the whole app).

As for running the full test suite, I think you're right in we shouldn't run all tests for each PR. I think we could do this with the test tagging and using the `--grep` argument for mocha, but I think that will need to be a separate PR.

**Comment by BenLambertNcl** [comment]
As discussed offline, AppVeyor UI tests will only run on master, which will significantly speed up merging of PRs.
